## ğŸ§¬ RNA-seq åˆ†ææµç¨‹ï¼ˆHISAT2 + StringTie + featureCountsï¼‰

 RNA-seq åˆ†ææµç¨‹
### ğŸ“¦ æµç¨‹æ¦‚è§ˆ


<img width="1201" height="225" alt="image" src="https://github.com/user-attachments/assets/89296ae4-5a60-484c-8ee4-6726b74ab92a" />



### ğŸ”§ æ¨¡å—è¯¦è§£

æ¨¡å—

è„šæœ¬

è¾“å…¥

æ ¸å¿ƒå·¥å…·

è¾“å‡º

è¯´æ˜

**1. è´¨é‡æ§åˆ¶**

`fp.sh`

`0.rawdata/*.fq.gz`

`fastp`

`1.fastp/results/1_qc/*.clean.fq.gz`

è‡ªåŠ¨å»æ¥å¤´ã€è´¨æ§ã€è¿‡æ»¤ä½è´¨é‡ reads

**2. åŸºå› ç»„ç´¢å¼•**

`his.sh`

`reference/*.fasta`

`hisat2-build`

`results/0_reference/hisat2_index/*.ht21`

ä¸€æ¬¡æ€§æ„å»ºï¼Œåç»­æ¯”å¯¹å¤ç”¨

**3. åºåˆ—æ¯”å¯¹**

`map.sh`

Clean FASTQ + ç´¢å¼•

`hisat2`,  `samtools`

`3.map/results/*.sorted.bam`

å¹¶è¡Œæ¯”å¯¹ + å®æ—¶æ’åº

**4.1 è½¬å½•æœ¬ç»„è£…**

`stringtie.sh`

`*.sorted.bam`

`stringtie`,  `parallel`

`4.stringtie/4.1results/*.gtf`

æ¯æ ·æœ¬ç‹¬ç«‹ç»„è£…è½¬å½•æœ¬ç»“æ„

**4.2 æ³¨é‡Šåˆå¹¶**

`merge.sh`

æ‰€æœ‰ sample.gtf

`stringtie --merge`

`4.stringtie/4.2results/merged_transcripts.gtf`

ç”Ÿæˆç»Ÿä¸€éå†—ä½™åŸºå› /è½¬å½•æœ¬æ³¨é‡Š

**4.3 è¡¨è¾¾å®šé‡**

`abd.sh`

BAM + merged.gtf

`featureCounts`,  `Python`

`5.counts/{raw_counts.tsv, tpm_counts.tsv}`

**åŸºå› æ°´å¹³ counts â†’ TPM**

----------


-   **åŒè¾“å‡º**ï¼š
    -   `raw_counts.tsv`  
    -   `tpm_counts.tsv`  
